carina-webdriver/src/main/java/com/qaprosoft/carina/core/foundation/webdriver/listener/DriverListener.java

```diff
@@ -254,7 +252,13 @@ private void captureScreenshot(String comment, WebDriver driver, WebElement elem
                 Screenshot.captureFailure(driver, comment); // in case of failure
             } else {
                 LOGGER.info(comment);
-                Screenshot.capture(driver, comment);
+                if (MobileContextHelper.isInWebViewContext()) {
+                    MobileContextHelper.backUpContext(driver);
+                    Screenshot.capture(driver, comment);
+                    MobileContextHelper.restoreContext(driver);
```

Comment:
I don't see sense from this  if (MobileContextHelper.isInWebViewContext())

Reply:
I have added a flag so I can check if the context was changed. I could change it to a call to the get context and return true if it contains WEBVIEW if you wish.